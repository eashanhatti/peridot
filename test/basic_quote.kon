let {
  val quo : (Quote '[A : OType] -> '[x : A] -> A) = (qlam \A { (qlam \x { x }) });
} in {
  (pure unit)
}
