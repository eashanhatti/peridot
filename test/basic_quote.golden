((fromList
    [(Global {unGlobal = 0}, Just (TypeType Object)),
     (Global {unGlobal = 1},
      Just
        (ObjectFunType
           (FreeVar (Level {unLevel = 0}))
           (Closure
              [FreeVar (Level {unLevel = 0})]
              (LocalVar (Index {unIndex = 1}))))),
     (Global {unGlobal = 2},
      Just (QuoteType (UniVar (Global {unGlobal = 0})))),
     (Global {unGlobal = 3},
      Just (QuoteType (UniVar (Global {unGlobal = 1})))),
     (Global {unGlobal = 4}, Nothing),
     (Global {unGlobal = 5}, Nothing),
     (Global {unGlobal = 6},
      Just (QuoteType (UniVar (Global {unGlobal = 4})))),
     (Global {unGlobal = 7},
      Just (QuoteType (UniVar (Global {unGlobal = 5})))),
     (Global {unGlobal = 8}, Just UnitType)],
  fromList [],
  fromList
    [(Global {unGlobal = 6}, Global {unGlobal = 7}),
     (Global {unGlobal = 7}, Global {unGlobal = 6})],
  []),
 Let
   [ObjTerm
      (Id {unId = 1})
      (QuoteType
         (ObjectFunType
            (TypeType Object)
            (ObjectFunType
               (LocalVar (Index {unIndex = 0}))
               (LocalVar (Index {unIndex = 1})))))
      (QuoteIntro
         (FunIntro
            (MetaFunIntro
               (QuoteIntro
                  (FunIntro
                     (MetaFunIntro (LocalVar (Index {unIndex = 0}))))))))]
   (IOIntroPure UnitIntro))
